src/fetch-wrapper.ts

```diff
@@ -153,7 +153,14 @@ export default function fetchWrapper(
 async function getResponseData(response: Response) {
   const contentType = response.headers.get("content-type");
   if (/application\/json/.test(contentType!)) {
-    return response.json();
+    // In the event that we get an empty response body we fallback to
+    // using .text(), but this should be investigated since if this were
+    // to occur in the GitHub API it really should not return an empty body.
+    try {
+      return response.json();
+    } catch (e) {
+      return response.text();
```

Comment:
```suggestion
      // istanbul ignore next - see comment above
      return response.text();
```

Reply:
given that [the `await`](https://github.com/octokit/request.js/pull/648/files#r1374843426) could easily get lost in a future refactoring, I think we should add a proper test for this case after all, instead of ignoring the case, to avoid future regression. Especially given that this is happening very sporadically, we wouldn't realize the regression for a longer time.